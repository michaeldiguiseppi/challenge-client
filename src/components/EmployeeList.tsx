import React from "react";
import EmployeeRow from "./EmployeeRow";
import type { Employee } from "./types";

type Props = {
  employees: Employee[];
};

const EmployeeList: React.FC<Props> = ({ employees }) => (
  <div>
    {employees.map((employee, index) => (
      <EmployeeRow employee={employee} key={index} />
    ))}
  </div>
);

export default EmployeeList;
