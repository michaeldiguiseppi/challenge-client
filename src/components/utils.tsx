import type { Employee } from "./types";

export const mockEmployees: Employee[] = [
  {
    name: {
      first: "Niclas",
      last: "Stornes",
      title: "Mr",
    },
    email: "niclas.stornes@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/78.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Erik",
      last: "Banks",
      title: "Mr",
    },
    email: "erik.banks@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/men/67.jpg",
      large: "https://randomuser.me/api/portraits/men/67.jpg",
      medium: "https://randomuser.me/api/portraits/med/men/67.jpg",
    },
    id: null,
  },
  {
    name: {
      first: "Andrea",
      last: "Crespo",
      title: "Mrs",
    },
    email: "andrea.crespo@example.com",
    picture: {
      thumbnail: "https://randomuser.me/api/portraits/thumb/women/41.jpg",
      large: "https://randomuser.me/api/portraits/women/41.jpg",
      medium: "https://randomuser.me/api/portraits/med/women/41.jpg",
    },
    id: null,
  },
];
