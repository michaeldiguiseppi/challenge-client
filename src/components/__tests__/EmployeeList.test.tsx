import React from "react";
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import EmployeeList from "../EmployeeList";
import { mockEmployees } from "../utils";

beforeEach(cleanup);

describe("<EmployeeList />", () => {
  it("renders without crashing", () => {
    render(
      <Router>
        <EmployeeList employees={mockEmployees} />
      </Router>
    );
  });

  it("should render a list of employees", () => {
    const { getByText } = render(
      <Router>
        <EmployeeList employees={mockEmployees} />
      </Router>
    );
    getByText(/niclas stornes/i);
  });

  it("should have the same number of rows as employees passed in", () => {
    const { getAllByTestId } = render(
      <Router>
        <EmployeeList employees={mockEmployees} />
      </Router>
    );
    const employeeRows = getAllByTestId(/employee-row/i);
    expect(employeeRows).toHaveLength(3);
  });
});
