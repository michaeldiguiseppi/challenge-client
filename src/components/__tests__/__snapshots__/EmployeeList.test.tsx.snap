// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmployeeList /> renders without crashing 1`] = `
<div>
  <div
    class="employee-list-container"
  >
    <a
      href="/people/niclas.stornes%40example.com"
    >
      <div
        class="employee-row"
        data-testid="employee-row"
      >
        <div
          class="employee-row-top-container"
        >
          <div
            class="employee-row-photo"
          >
            <img
              alt="Niclas-Stornes"
              src="https://randomuser.me/api/portraits/thumb/men/78.jpg"
            />
          </div>
          <div
            class="employee-row-name"
          >
            Mr. Niclas Stornes
          </div>
        </div>
      </div>
    </a>
    <a
      href="/people/erik.banks%40example.com"
    >
      <div
        class="employee-row"
        data-testid="employee-row"
      >
        <div
          class="employee-row-top-container"
        >
          <div
            class="employee-row-photo"
          >
            <img
              alt="Erik-Banks"
              src="https://randomuser.me/api/portraits/thumb/men/67.jpg"
            />
          </div>
          <div
            class="employee-row-name"
          >
            Mr. Erik Banks
          </div>
        </div>
      </div>
    </a>
    <a
      href="/people/andrea.crespo%40example.com"
    >
      <div
        class="employee-row"
        data-testid="employee-row"
      >
        <div
          class="employee-row-top-container"
        >
          <div
            class="employee-row-photo"
          >
            <img
              alt="Andrea-Crespo"
              src="https://randomuser.me/api/portraits/thumb/women/41.jpg"
            />
          </div>
          <div
            class="employee-row-name"
          >
            Mrs. Andrea Crespo
          </div>
        </div>
      </div>
    </a>
  </div>
</div>
`;
